ts/src/index.ts

```diff
@@ -118,6 +136,17 @@ async function initConfigMetadata(config: ProfilerConfig):
       config.instance = instance;
     }
   }
+  if (!config.serviceContext.service) {
+    try {
+      config.serviceContext.service =
+          await getMetadataInstanceField('attributes/cluster-name');
```

Comment:
I am not convinced this is useful. The cluster typically runs many microservices (pods in Kubernetes speak). Aggregating all of them together is not useful.

Reply:
Is there a way to figure out the pod name? I agree that would be better, and we should probably fix that in the debug agent as well.